src/common/k8s/resource-stack.ts

```diff
@@ -0,0 +1,152 @@
+   */
+  async kubectlApplyFolder(folderPath: string, templateContext?: any, extraArgs?: string[]): Promise<string> {
+    const resources = await this.renderTemplates(folderPath, templateContext);
+
+    return this.applyResources(resources, extraArgs);
+  }
+
+  /**
+   *
+   * @param folderPath folder path that is searched for files defining kubernetes resources.
+   * @param templateContext sets the template parameters that are to be applied to any templated kubernetes resources that are to be applied.
+   */
+  async kubectlDeleteFolder(folderPath: string, templateContext?: any, extraArgs?: string[]): Promise<string> {
+    const resources = await this.renderTemplates(folderPath, templateContext);
+
+    return this.deleteResources(resources, extraArgs);
+  }
+
+  protected async applyResources(resources: string[], extraArgs?: string[]): Promise<string> {
+    const clusterModel = ClusterStore.getInstance().getById(this.cluster.metadata.uid);
+
+    if (!clusterModel) {
+      throw new Error(`cluster not found`);
+    }
+
+    let kubectlArgs = extraArgs || [];
+
+    kubectlArgs = this.appendKubectlArgs(kubectlArgs);
+
+    if (app) {
+      return await new ResourceApplier(clusterModel).kubectlApplyAll(resources, kubectlArgs);
+    } else {
+      const response = await requestMain(clusterKubectlApplyAllHandler, this.cluster.metadata.uid, resources, kubectlArgs);
+
+      if (response.stderr) {
+        throw new Error(response.stderr);
+      }
+
+      return response.stdout;
+    }
+  }
+
+  protected async deleteResources(resources: string[], extraArgs?: string[]): Promise<string> {
+    const clusterModel = ClusterStore.getInstance().getById(this.cluster.metadata.uid);
+
+    if (!clusterModel) {
+      throw new Error(`cluster not found`);
+    }
+
+    let kubectlArgs = extraArgs || [];
+
+    kubectlArgs = this.appendKubectlArgs(kubectlArgs);
+
+    if (app) {
+      return await new ResourceApplier(clusterModel).kubectlDeleteAll(resources, kubectlArgs);
+    } else {
+      const response = await requestMain(clusterKubectlDeleteAllHandler, this.cluster.metadata.uid, resources, kubectlArgs);
+
+      if (response.stderr) {
+        throw new Error(response.stderr);
+      }
+
+      return response.stdout;
+    }
+  }
+
+  protected appendKubectlArgs(kubectlArgs: string[]) {
+    if (!kubectlArgs.includes("-l") && !kubectlArgs.includes("--label")) {
+      return kubectlArgs.concat(["-l", `app.kubernetes.io/name=${this.name}`]);
+    }
+
+    return kubectlArgs;
+  }
+
+  protected async renderTemplates(folderPath: string, templateContext: any): Promise<string[]> {
+    const resources: string[] = [];
+
+    logger.info(`[RESOURCE-STACK]: render templates from ${folderPath}`);
+    const files = await fse.readdir(folderPath);
+
+    for(const filename of files) {
+      const file = path.join(folderPath, filename);
+      const raw = await fse.readFile(file);
```

Comment:
nit: since we always `.toString()` this, we might as well declare the expected encoding and get `readFile()` to do it for us.
```suggestion
      const raw = await fse.readFile(file, "utf-8");
```